plugins/CoreHome/vue/src/NumberFormatter/NumberFormatter.ts

```diff
@@ -0,0 +1,249 @@
+      ',': Matomo.numbers.symbolGroup,
+      '+': Matomo.numbers.symbolPlus,
+      '-': Matomo.numbers.symbolMinus,
+      '%': Matomo.numbers.symbolPercent,
+    };
+
+    let newValue = '';
+    const valueParts = value.split('');
+
+    valueParts.forEach((val) => {
+      let valueReplaced = val;
+
+      Object.entries(replacements).some(([char, replacement]) => {
+        if (valueReplaced.indexOf(char) !== -1) {
+          valueReplaced = valueReplaced.replace(char, replacement);
+          return true;
+        }
+
+        return false;
+      });
+
+      newValue += valueReplaced;
+    });
+
+    return newValue;
+  }
+
+  private valOrDefault(val: number|undefined, def: number): number {
+    if (typeof val === 'undefined') {
+      return def;
+    }
+
+    return val;
+  }
+
+  public parseFormattedNumber(value: string): number {
+    const isNegative = value.indexOf(Matomo.numbers.symbolMinus) > -1 || value.startsWith('-');
+    const numberParts = value.split(Matomo.numbers.symbolDecimal);
+
+    numberParts.forEach((val, index) => {
+      numberParts[index] = val.replace(/[^0-9]/g, '');
+    });
+
+    return (isNegative ? -1 : 1) * parseFloat(numberParts.join('.'));
+  }
+
+  public formatNumber(
+    value: string|number,
+    maxFractionDigits?: number,
+    minFractionDigits?: number,
+  ): string {
+    return this.format(
+      value,
+      Matomo.numbers.patternNumber,
+      this.valOrDefault(maxFractionDigits, this.defaultMaxFractionDigits),
+      this.valOrDefault(minFractionDigits, this.defaultMinFractionDigits),
+    );
+  }
+
+  public formatPercent(
+    value: string|number,
+    maxFractionDigits?: number,
+    minFractionDigits?: number,
+  ): string {
+    return this.format(
+      value,
+      Matomo.numbers.patternPercent,
+      this.valOrDefault(maxFractionDigits, this.defaultMaxFractionDigits),
+      this.valOrDefault(minFractionDigits, this.defaultMinFractionDigits),
+    );
+  }
+
+  public formatCurrency(
+    value: string|number,
+    currency: string,
+    maxFractionDigits?: number,
+    minFractionDigits?: number,
+  ): string {
+    const formatted = this.format(
+      value,
+      Matomo.numbers.patternCurrency,
+      this.valOrDefault(maxFractionDigits, this.defaultMaxFractionDigits),
+      this.valOrDefault(minFractionDigits, this.defaultMinFractionDigits),
+    );
+    return formatted.replace('¤', currency);
+  }
+
+  public formatEvolution(
+    evolution: string|number,
+    maxFractionDigits?: number,
+    minFractionDigits?: number,
+    noSign?: boolean,
+  ): string {
+    if (noSign) {
+      return this.formatPercent(
+        Math.abs(evolution as number),
+        maxFractionDigits,
+        minFractionDigits,
+      );
+    }
+    const formattedEvolution = this.formatPercent(evolution, maxFractionDigits, minFractionDigits);
+    return `${evolution as number > 0 ? Matomo.numbers.symbolPlus : ''}${formattedEvolution}`;
+  }
+
+  public calculateAndFormatEvolution(
```

Comment:
There is something missing in JS:

https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/plugins/CoreHome/Columns/Metrics/EvolutionMetric.php#L240

The precomputed metrics are calculated with a ratio adjustment for incomplete periods, e.g. the current year get a factor applied to show a besser "estimated" evolution before the year ended.

And with JS potentially not understanding the PHP timezones (so a day can be adjusted with the correct ratio), I assume we need to add something like `Date::getUtcOffset(Site::getTimezoneFor($idSite))` in `Dashboard::enrichValues()`. And additionally pass the `ts_archived` value from the API as that affects the final ratio.

Or we use the ratio we already receive from the API. :thinking:

Reply:
I've for now used the `ratio` value. We can check if we want to change that when cleaning up all the code after removing the old all websites dashboard.